From 22eebbf0b5ab380173b3d8bab558719b1f408a2e Mon Sep 17 00:00:00 2001
From: Daniel Mack <zonque@gmail.com>
Date: Thu, 20 Feb 2014 13:29:54 +0100
Subject: [PATCH] core/unit: fix unit_add_target_dependencies() for units with
 no dependencies

For units without any dependencies, r needs to be initialized to 0.
Otherwise, the return value of unit_add_target_dependencies() is
unspecified.

Conflicts:
	src/core/unit.c

(cherry-picked from db57f3c6d11864f4227bb6734a0d509695a32a7d)

Resolves: #1147524
---
 src/core/unit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index e56971b..b51e351 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -923,8 +923,8 @@ static int unit_add_default_dependencies(Unit *u) {
 
         Unit *target;
         Iterator i;
-        int r;
         unsigned k;
+        int r = 0;
 
         assert(u);
 
